### DISCUSSION TITLE
Volume reducing after passing allpass filter

### DISCUSSION THREAD

**Discussion by puneet-yadav** [katspaugh/wavesurfer.js#3834] [Q&A]
If I play the file without giving any filters the , the volume seems to be loud. But if I apply filter - allpass. The file does not sound the same. I have set the backend as MediaElement. 
But if I set the backend as WebAudio. File sounds same as without the filters as it will sound in all pass.

This is my code

const createAudioContext = () => {
    if (audioContext.current === null) {
      audioContext.current = new AudioContext();
      mediaNode.current = audioContext.current.createMediaElementSource(
        wavesurfer.current.getMediaElement()
      );
    }
  };

  const applyFilterSettings = () => {
    createAudioContext();

    mediaNode.current.disconnect();

    const filter = audioContext.current.createBiquadFilter();
    filter.type = filterType.value;
    let filterAudioValue = parseInt(audioFilterValue);
    filter.frequency.value = filterAudioValue;

    mediaNode.current.connect(filter);
    filter.connect(audioContext.current.destination);
    
  };

Please help. @katspaugh

**Comment by katspaugh** [comment]
What's your question exactly?

**Reply by puneet-yadav** [reply]
I want to apply Filters to my sound wave. I refered to the discussion -> https://github.com/katspaugh/wavesurfer.js/discussions/3415
I have two questions  
1. is it necessary to give the backend as WebAudio if I want to apply filters ( low, high, band).
2. If I give give MediaElement to backend and play sound without filters it appears loud. but if I apply allpass filter , the sound should come same ? it is not coming same. So basically how can I remove the filters applied to have the same sound which I had when no filter was applied

**Reply by katspaugh** [reply]
1. No. In fact, it will not work the same way as with a media element.
2. If you apply a filter to audio, by definition the sound won't be the same. A filter cuts off a certain range of frequencies.

**Reply by puneet-yadav** [reply]
How can I remove the applied filter
.

**Reply by katspaugh** [reply]
It's a Web Audio question, not wavesurfer. Try ChatGPT or Google.
